### ISSUE TITLE
tomlkit 0.5.9 breaking change?

### ISSUE THREAD

**Issue by jonapich** [sdispater/tomlkit#78]
It was working fine before, but not anymore:

```
+ poetry install
(...)
[RuntimeError]
The Poetry configuration is invalid:
  - {...} is not of type 'object'
```

I have compared the `pip install poetry` outputs of both working and non-working versions. The only difference is tomlkit 0.5.8 -> 0.5.9

I have changed my CI command line to do `pip install poetry "tomlkit=0.5.8"` and it's working fine again.

Let me know if you need more information about my project setup.

**Comment by sdispater** [comment]
Can you please provide the `pyproject.toml` file that causes the issue?

**Comment by jonapich** [comment]
```
[tool.poetry]
name = "scripts"
version = "0.0.1"
description = "..."
authors = ["jonapich <jonapich@nicetry.com>"]

[tool.poetry.dependencies]
python = ">=3.6"

click = "*"
emoji = "*"
fsutils = { path = '../fsutils/'}
typing_extensions = "*"


[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"


[tool.poetry.scripts]
jid2python = "scripts.fsutils:fsutils"
package-errorlog-detector = "scripts.package_errorlog_detector:package_errorlog_detector"
pypi = "scripts.pypi:pypi"
emoji = "scripts.emoji:emoji"
```

**Comment by jonapich** [comment]
Another detail: the pyproject above is loaded through a path reference. I use a pyproject.toml at the root of my repo to tie several different python packages together just for development.

The root pyproject looks like this:

```
[tool.poetry]
...

[tool.poetry.dependencies]
python = ">=3.6"

scripts = { path = "some_path/scripts/" }


[tool.poetry.dev-dependencies]
mypy = "*"
tests = { path = "some_path/tests/" }


[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"
```

**Comment by sdispater** [comment]
The issue is that you have out of order tables `[build-system]` is between `[tool.poetry]` and `[tool.poetry.scripts]`. If you put `[tool.poetry.scripts]` before `[build-system]`, the error should go away.

**Comment by sdispater** [comment]
But this is still an issue with `tomlkit` introduced in #68

**Comment by jonapich** [comment]
oooh I never realized the order was a requirement... thanks

**Comment by sdispater** [comment]
The order shouldn't be a requirement though since the TOML specification authorizes it.

I will make a fix.

**Comment by jonapich** [comment]
Thank you for the awesome turnaround speed.

**Comment by sdispater** [comment]
I just released the `0.5.10` version which should fix the issue.
